provider/dataset/materialization_dataset.go

```diff
@@ -0,0 +1,98 @@
+package dataset
+
+import (
+	"context"
+
+	types "github.com/featureform/fftypes"
+)
+
+type SizedSegmentableChunkedDataset interface {
+	SizedSegmentableDataset
+	ChunkedDataset
+}
+
+type MaterializationID string
+
+type FeatureIterator interface {
```

Comment:
Arguably no point in having these things be interfaces, just make the Impls the actual structs